File: javascript/leetcode/find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    seen = {}
    dupes = []
    for (var i = 0; i < nums.length; i++) {
        if (seen[nums[i]]) {
            dupes.push(nums[i])
        } else {
            seen[nums[i]] = 1
        }
    }
    return dupes
};

File: python/leetcode/find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        seen = {}
        dupes = []
        for i in range(len(nums)):
            if nums[i] in seen:
                dupes.append(nums[i])
            else:
                seen[nums[i]] = 1
        return dupes
